### ISSUE TITLE
fakegooglebot errors on Ubuntu 16.04

### ISSUE THREAD

**Issue by junior466** [fail2ban/fail2ban#2419]
### Environment:

_Fill out and check (`[x]`) the boxes which apply. If your Fail2Ban version is outdated, 
and you can't verify that the issue persists in the recent release, better seek support 
from the distribution you obtained Fail2Ban from_

- Fail2Ban version (including any possible distribution suffixes): Fail2Ban v0.9.3
- OS, including release name/version: Ubuntu 16.04
- [ x] Fail2Ban installed via OS/distribution mechanisms
- [ x] You have not applied any additional foreign patches to the codebase
- [ ] Some customizations were done to the configuration (provide details below is so)

### The issue:

```
 [19339]: ERROR   /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot IP_HERE -- stderr: b'Traceback (most recent call last):\n  
File "/etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot", line 32, in <module>\n    is_googlebot(process_args(sys.argv))\n  File "/etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot", line 17, in process_args\n    from fail2ban.server.filter import DNSUtils\nImportError: No module named fail2ban.server.filter\n'
fail2ban.action         [19339]: ERROR   /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot IP_HERE -- returned 1
```

I didn't add any more information because I believe this to be a duplicate of [this](https://github.com/fail2ban/fail2ban/issues/2293) but as a beginner, I don't quite understand how to fix this issue exactly. 

I did run this command `ls -l /usr/bin/python*` as a troubleshooting step and this is the result:

```
lrwxrwxrwx 1 root root       9 Nov 23  2017 /usr/bin/python -> python2.7
lrwxrwxrwx 1 root root       9 Nov 23  2017 /usr/bin/python2 -> python2.7
-rwxr-xr-x 1 root root 3492656 Nov 12 14:46 /usr/bin/python2.7
lrwxrwxrwx 1 root root       9 May  5  2018 /usr/bin/python3 -> python3.5
-rwxr-xr-x 2 root root 4464368 Nov 12 11:27 /usr/bin/python3.5
-rwxr-xr-x 2 root root 4464368 Nov 12 11:27 /usr/bin/python3.5m
-rwxr-xr-x 1 root root     384 Feb  5  2018 /usr/bin/python3-futurize
lrwxrwxrwx 1 root root      10 May  5  2018 /usr/bin/python3m -> python3.5m
-rwxr-xr-x 1 root root     388 Feb  5  2018 /usr/bin/python3-pasteurize
-rwxr-xr-x 1 root root     152 Nov 23  2015 /usr/bin/python3-pbr
```

Any help would be greatly appreciated.

**Comment by sebres** [comment]
See https://github.com/fail2ban/fail2ban/issues/2293#issuecomment-445207038
Closed as fixed / duplicate.

**Comment by sebres** [comment]
> I don't quite understand how to fix this issue exactly.

You could test within which python version fail2ban was installed using something like this:
```
python2.7 -c 'from fail2ban.server.filter import DNSUtils'
python3.5 -c 'from fail2ban.server.filter import DNSUtils'
```
If you found that, just change the shebang in `/etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot` (open it in editor your choice, modify as noticed in comment and save).
To test it works, just call in shell (it should not produce the error `No module named fail2ban.server.filter` anymore):
```
/etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 192.2.0.1
```

**Comment by junior466** [comment]
@sebres So all I have to change is 

`#!/usr/bin/python
# Inspired by https://isc.sans.edu/forums/diary/When+Google+isnt+Google/15968/
#
# Written in Python to reuse built-in Python batteries and not depend on
# presence of host and cut commands
#`

to 

`#!/usr/bin/python3.5
# Inspired by https://isc.sans.edu/forums/diary/When+Google+isnt+Google/15968/
#
# Written in Python to reuse built-in Python batteries and not depend on
# presence of host and cut commands
#
`

If so, then then `/etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 192.2.0.1` no longer displays an error :-)

Thank you!

**Comment by junior466** [comment]
@sebres Forgot to add that now I see:

```
2019-05-10 13:35:18,731 fail2ban.action         [27253]: ERROR   /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 108.162.221.29 -- stdout: b''
2019-05-10 13:35:18,731 fail2ban.action         [27253]: ERROR   /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 108.162.221.29 -- stderr: b''
2019-05-10 13:35:18,731 fail2ban.action         [27253]: ERROR   /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 108.162.221.29 -- returned 1
```

I believe to be the same as [this](https://github.com/fail2ban/fail2ban/issues/1194).  Based on that thread, the fix was only implemented in the 0.10th branch?

**Comment by sebres** [comment]
> Based on that thread, the fix was only implemented in the 0.10th branch?

Yes... Is it not obvious?
One could try to cherry-pick this to 0.9 (but simply too different, so I have to pass)...

Please try to understand that we haven't interest to merge back all the things of v.0.10 to v.0.9. Just because then 0.9 will be simply 0.10.
Especially if some fixes are pretty simply in 0.10 and would cost a lot of effort for back-porting to 0.9.

Your version is outdated (and further development is discontinued).
One could update to github-version, or ... find some repository with newest version, or ... wait for the maintainers would release new version for your distribution.
